facet-core/src/impls_core/ops.rs

```diff
@@ -0,0 +1,64 @@
+use crate::{
+    ConstTypeId, Def, Facet, Field, Shape, StructDef, StructKind, VTableView, ValueVTable,
+};
+use core::{alloc::Layout, mem};
+
+unsafe impl<'a, Idx: Facet<'a>> Facet<'a> for core::ops::Range<Idx> {
+    const SHAPE: &'static crate::Shape = &const {
+        Shape::builder_for_sized::<Self>()
+            .type_params(&[crate::TypeParam {
+                name: "Idx",
+                shape: || Idx::SHAPE,
+            }])
+            .id(ConstTypeId::of::<Self>())
+            .layout(Layout::new::<Self>())
+            .vtable(
+                &const {
+                    let mut builder = ValueVTable::builder::<Self>().type_name(|f, opts| {
+                        write!(f, "Range")?;
+                        if let Some(opts) = opts.for_children() {
+                            write!(f, "<")?;
+                            (Idx::SHAPE.vtable.type_name)(f, opts)?;
+                            write!(f, ">")?;
+                        } else {
+                            write!(f, "<…>")?;
+                        }
+                        Ok(())
+                    });
+
+                    if Idx::SHAPE.is_debug() {
+                        builder = builder.debug(|this, f| {
+                            (<VTableView<Idx>>::of().debug().unwrap())(&this.start, f)?;
+                            write!(f, "..")?;
+                            (<VTableView<Idx>>::of().debug().unwrap())(&this.end, f)?;
+                            Ok(())
+                        });
+                    }
```

Comment:
I'm curious what's actually the purpose of reimplementing Debug when facet-pretty isn't even using these impls? Also how would one call this debug impl?